src/iterative_ensemble_smoother/esmda.py

```diff
@@ -279,9 +281,11 @@ def assimilate(
         # if C_D = L L.T by the cholesky factorization, then drawing y from
         # a zero cented normal means that y := L @ z, where z ~ norm(0, 1)
         # Therefore, scaling C_D by alpha is equivalent to scaling L with sqrt(alpha)
-        D = self.perturb_observations(
-            ensemble_size=num_ensemble, alpha=self.alpha[self.iteration]
-        )
+        if D is None:
```

Comment:
I guess this does not need to be there?

Reply:
This test on `D is None` is there for ESMDA to itself calculate `D `instead of having` D` as input. For backward compatibility `D `is not assigned but calculated inside ESMDA class.

Reply:
Thanks for clarification. I think then we can keep it then - to have the possibility to get observations from outside 👍